library/items/item/src/main/java/org/quiltmc/qsl/item/api/item/v1/QuiltItemSettings.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.item.api.item.v1;
+
+import net.minecraft.item.FoodComponent;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Rarity;
+
+import org.quiltmc.qsl.item.impl.QuiltItemInternals;
+
+/**
+ * Quilts's version of {@link Item.Settings}. Adds additional methods and hooks
+ * not found in the original class.
+ *
+ * <p>To use it, simply replace {@code new Item.Settings()} with
+ * {@code new QuiltItemSettings()}.
+ */
+public class QuiltItemSettings extends Item.Settings {
```

Comment:
This would probably be as good a time as any to add support for [custom item settings](https://github.com/FabricMC/fabric/pull/1359).